### ISSUE TITLE
[engine] caught signal (SIGTERM)

### ISSUE THREAD

**Issue by rohitrk89** [fluent/fluent-bit#7927]
### Discussed in https://github.com/fluent/fluent-bit/discussions/7926

<div type='discussions-op-text'>

<sup>Originally posted by **rohitrk89** September 15, 2023</sup>
We have deployed Fluentbit in our EKS stack to push the logs to s3 bucket.
in one of the env configs are working fine, however in another pods are crashing.
Below are the logs generated before pod is crashed.

```
Fluent Bit v1.8.9
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2023/09/15 05:35:55] [ info] [engine] started (pid=1)
[2023/09/15 05:35:55] [ info] [storage] version=1.1.5, initializing...
[2023/09/15 05:35:55] [ info] [storage] in-memory
[2023/09/15 05:35:55] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2023/09/15 05:35:55] [ info] [cmetrics] version=0.2.2
[2023/09/15 05:35:55] [ info] [filter:kubernetes:kubernetes.0] https=1 host=kubernetes.default.svc port=443
[2023/09/15 05:35:55] [ info] [filter:kubernetes:kubernetes.0] local POD info OK
[2023/09/15 05:35:55] [ info] [filter:kubernetes:kubernetes.0] testing connectivity with API server...
[2023/09/15 05:35:55] [ info] [filter:kubernetes:kubernetes.0] connectivity OK
[2023/09/15 05:35:55] [ info] [fstore] created root path /tmp/fluent-bit/s3/optus-dpm-acma-dev-mfa-data
[2023/09/15 05:35:55] [ info] [output:s3:s3.0] Using upload size 1000000 bytes
[2023/09/15 05:35:55] [ info] [output:s3:s3.0] total_file_size is less than 10 MB, will use PutObject API
[2023/09/15 05:36:25] [engine] caught signal (SIGTERM)
[2023/09/15 05:36:25] [ info] [input] pausing tail.0

```
kubectl event/s 
```
31m         Normal    Created            pod/fluentbit-s3-fluent-bit-p24zf   Created container fluent-bit
31m         Normal    Started            pod/fluentbit-s3-fluent-bit-p24zf   Started container fluent-bit
31m         Warning   Unhealthy          pod/fluentbit-s3-fluent-bit-p24zf   Readiness probe failed: Get "http://10.121.82.177:2020/api/v1/health": dial tcp 10.121.82.177:2020: connect: connection refused
6m52s       Warning   Unhealthy          pod/fluentbit-s3-fluent-bit-p24zf   Liveness probe failed: Get "http://10.121.82.177:2020/": dial tcp 10.121.82.177:2020: connect: connection refused
31m         Normal    Killing            pod/fluentbit-s3-fluent-bit-p24zf   Container fluent-bit failed liveness probe, will be restarted
111s        Warning   BackOff            pod/fluentbit-s3-fluent-bit-p24zf   Back-off restarting failed container
38m         Warning   Unhealthy          pod/fluentbit-s3-fluent-bit-q5jhv   Readiness probe failed: Get "http://10.121.83.118:2020/api/v1/health": dial tcp 10.121.83.118:2020: connect: connection refused
43m         Warning   BackOff            pod/fluentbit-s3-fluent-bit-q5jhv   Back-off restarting failed container

```

very new to this implementation, any help will be appreciated.
 </div>

**Comment by patrick-stephens** [comment]
This version is very old and no longer maintained. Please upgrade to the latest 2.1 series and re-test to confirm it is not already resolved?

**Comment by nokute78** [comment]
In addition, could you share configuration file and enable http server ?

```
31m         Warning   Unhealthy          pod/fluentbit-s3-fluent-bit-p24zf   Readiness probe failed: Get "http://10.121.82.177:2020/api/v1/health": dial tcp 10.121.82.177:2020: connect: connection refused
6m52s       Warning   Unhealthy          pod/fluentbit-s3-fluent-bit-p24zf   Liveness probe failed: Get "http://10.121.82.177:2020/": dial tcp 10.121.82.177:2020: connect: connection refused
31m         Normal    Killing            pod/fluentbit-s3-fluent-bit-p24zf   Container fluent-bit failed liveness probe, will be restarted
```

**Comment by rohitrk89** [comment]
Thanks @patrick-stephens  and @nokute78 for you help, I managed to fix the issue with version upgrade.

**Comment by rohitrk89** [comment]
issue fixed post upgrade to 2.1.9
